tests/integration_test/dns_resolver_test.go

```diff
@@ -25,63 +25,85 @@ import (
 	"github.com/GoogleCloudPlatform/esp-v2/tests/env/platform"
 )
 
-func fullDns(dns string) string {
+func toFqdnWithRoot(dns string) string {
 	return dns + "."
 }
 
 func TestDnsResolver(t *testing.T) {
 	t.Parallel()
-	s := env.NewTestEnv(comp.TestDnsResolver, platform.EchoSidecar)
-
-	// Setup dns resolver.
-	backendHost := "dns-resolver-test-backend"
-	localHost := "127.0.0.1"
-	dnsRecords := map[string]string{
-		fullDns(backendHost): localHost,
-	}
-	dnsResolver := comp.NewDnsResolver(s.Ports().DnsResolverPort, dnsRecords)
-	go func() {
-		if err := dnsResolver.ListenAndServe(); err != nil {
-			t.Fatalf("Failed to set udp listener %s\n", err.Error())
-		}
-	}()
-
-	// Check dns resolver's health.
-	dnsResolverAddress := fmt.Sprintf("127.0.0.1:%v", s.Ports().DnsResolverPort)
-	if err := comp.CheckDnsResolverHealth(dnsResolverAddress, backendHost, localHost); err != nil {
-		t.Fatalf("DNS Resolver is not healthy: %v", err)
-	}
-
-	// Setup the whole test framework.
-	s.SetBackendAddress(fmt.Sprintf("http://%s:%v", backendHost, s.Ports().BackendServerPort))
-	args := []string{"--service_config_id=test-config-id",
-		"--rollout_strategy=fixed", "--healthz=/healthz", "--dns_resolver_addresses=" + dnsResolverAddress}
 
 	testCase := []struct {
 		desc            string
+		backendHost     string
 		isResolveFailed bool
 		wantResp        string
 		wantError       string
 	}{
 		{
-			desc:     "resolve domain name successfully",
-			wantResp: `{"message":"hello"}`,
+			desc:        "resolve non-FQDN domain name successfully",
```

Comment:
IIUC, PQDN is tested here so maybe replace non-FQDN with PQDN.
http://www.tcpipguide.com/free/t_AbsoluteFullyQualifiedandRelativePartiallyQualifie.htm

Reply:
Done